server/src/main/java/io/crate/expression/operator/any/AnyNeqOperator.java

```diff
@@ -68,7 +68,7 @@ protected Query refMatchesAnyArrayLiteral(Function any, Reference probe, Literal
             if (value == null) {
                 continue;
             }
-            andBuilder.add(EqOperator.fromPrimitive(probe.valueType(), probe.column().fqn(), value), BooleanClause.Occur.MUST);
+            andBuilder.add(EqOperator.fromPrimitive(probe.valueType(), columnName, value), BooleanClause.Occur.MUST);
```

Comment:
Thanks for catching!

I think we need a test here. It was not caught by testNeqAnyOnArrayLiteral because on LuceneQueryBuilderTest-s we don't assign OID-s and thus probe.storageIdent() ==  probe.column().fqn().

Can we have a LuceneQueryBuilderIntegrationTest case for that?

Reply:
I don't think we want to add additional integration tests for all the query variants either.

We could have additional unit tests that test the query matching using the `QueryTester` without asserting the query output. Similar to `test_is_null_on_columns_without_doc_values`  (minus the _field_names stuff)

Reply:
As Baur said, with unit tests, I think we cannot determine whether we used `ref.storageIdent()` or `ref.column().fqn()` to generate Lucene queries since both return the column name (not oid).
For example, if we intentionally change https://github.com/crate/crate/blob/bd44a8890c7754d0159458065b36d8bad93983d5/server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java#L119 back to `ref.column().fqn()`, `test_is_null_on_columns_without_doc_values` will still pass.

Reply:
We could add a unit test variant which uses a `QueryTester` instance that uses a real OID supplier rather than the default one defined here https://github.com/crate/crate/blob/master/server/src/testFixtures/java/io/crate/testing/QueryTester.java#L89.

Reply:
Thank you, added a test but just to this specific case. I will expand to cover other `toQuery` implementations.

Reply:
>I will expand to cover other toQuery implementations.

We probably need to re-use `queryTester`. 
Add API to set and pass through to the `queryTester`, reset after each test to unassigned.

Also, it looks like we always use single column in tests, ie supplier returning same value might be okay. 

However, we might consider having a supplier with actual increment (all columns in `LuceneQueryBuilderTest.prepare` will get different oids) and adjust tests accordingly - ie update assertions to use different oids across tests cases and not always "1"

Reply:
> However, we might consider having a supplier with actual increment (all columns in LuceneQueryBuilderTest.prepare will get different oids) and adjust tests accordingly - ie update assertions to use different oids across tests cases. Just to be safe and be able to use 2+ columns in the same test.

Thanks I went this direction to let all tests to take effect and actually found another missing conversion from `NumTermsPerDocQuery`.